cirq-google/cirq_google/workflow/qubit_placement.py

```diff
@@ -107,6 +108,72 @@ def default_topo_node_to_qubit(node: Any) -> cirq.Qid:
         return cirq.LineQubit(node)
 
 
+class HardcodedQubitPlacer(QubitPlacer):
+    def __init__(
+        self,
+        mapping: Dict[cirq.NamedTopology, Dict[Any, cirq.Qid]],
+        topo_node_to_qubit_func: Callable[[Hashable], cirq.Qid] = default_topo_node_to_qubit,
+    ):
+        self.mapping = mapping
+        self.topo_node_to_qubit_func = topo_node_to_qubit_func
```

Comment:
`NaiveQubitPlacer` is a frozen dataclass, while other qubit placers have public objects, like `self.mapping` here. 

To keep things consistent (i.e. all qubit placers should be immutable), should we make sure that all internal objects are private (`self._mapping` instead of `self.mapping`) and we expose them via properties, if needed (thereby enforcing immutability)?

Reply:
yes, good idea.

Reply:
I'll make mapping private but it's still fairly mutable if the caller keeps a reference to the dictionary they passed to the constructor. I guess I could copy in the dictionary in the constructor, but I don't know if this is worth the overhead.